File: app/schemas/exchange.py
from pydantic import BaseModel
from uuid import UUID
from app.enums import ExchangeStatusEnum
from datetime import datetime
class ExchangeCreate(BaseModel):
    requested_book_id: UUID


class ExchangeResponse(BaseModel):
    id: UUID
    requester_id: UUID
    owner_id: UUID
    requester_name: str
    owner_name: str

    requested_book_id: UUID
    status: ExchangeStatusEnum

    created_at: datetime
    updated_at: datetime


    class Config:
        from_attributes = True

File: book-exchange-frontend/types/exchange.ts
export type ExchangeStatus =
  | "PENDING"
  | "ACCEPTED"
  | "REJECTED"
  | "CANCELLED"
  | "COMPLETED"
  | "WAITING_CONFIRMATION";

export interface Exchange {
  updated_at: string | number | Date;
  created_at: string | number | Date;
  id: string;
  requester_id: string;
  owner_id: string;
  requested_book_id: string;
  requester_name: string;
  owner_name: string;
  owner_confirmed: string;
  requester_confirmed: string;
  first_confirmed_at: string | number | Date;
  status: ExchangeStatus;
  
}
